components/gitpod-protocol/src/protocol.ts

```diff
@@ -595,6 +595,8 @@ export interface WorkspaceImageSourceDocker {
     dockerFileHash: string
     dockerFileSource: Commit
     dockerFilePath: string
+    /** The FROM part in the Dockerfile (optional). */
+    dockerFileFrom: string | undefined
```

Comment:
Random thought: If you include the resolved upstream hash in the `dockerfileFileHash` function, I think you won't need to modify protocol shapes, nor will you have to remember to compare `dockerFileFrom` in all relevant places.

Reply:
That's indeed an interesting idea. However, the problem is that we need to resolve the FROM image somewhere (to get the revision SHA hash) and as far as I know, we cannot resolve this image in the server component easily but in the image-builder component.